base/dates/io.jl

```diff
@@ -100,12 +100,12 @@ slotparse(slot::Slot{Millisecond},x) = !ismatch(r"[^0-9\s]",x) ? slot.period(Bas
 slotparse(slot::Slot{DayOfWeekSlot},x) = nothing
 
 function getslot(x,slot::DelimitedSlot,df,cursor)
-    endind = first(search(x,df.trans[slot.i],cursor+1))
+    endind = first(search(x,df.trans[slot.i],nextind(x,cursor)))
```

Comment:
No, this line was assuming that the string it index was ASCII, while what we want is the next Unicode codepoint. (This code is not working with the internals of the string, it just needs to access its characters via the high-level API.)